backend/sim/channel/asset.go

```diff
@@ -46,5 +47,14 @@ func (a Asset) MarshalBinary() ([]byte, error) {
 
 // Decode decodes a sim Asset from the io.Reader `r`.
 func (a *Asset) Decode(r io.Reader) error {
+	var length int64
```

Comment:
`int64`? why not `uint64`? could we use something shorter here, like `uint16`?

Reply:
Updated to use uint16 in b6137b1cdfaa284311fdb26a48d38c0ce24f1508. However, this commit will not be updated. Because, the decoding part is moved to `pkg/io` when introducing BinaryUnmarshaler in the later commit.